src/cli/rustup_mode.rs

```diff
@@ -130,11 +140,15 @@ pub fn cli() -> App<'static, 'static> {
                 .long("quiet"),
         )
         .arg(
-            Arg::with_name("toolchain")
+            Arg::with_name("toolchain-option")
```

Comment:
Why not just override `toolchain` instead of adding both `toolchain-option` and `toolchain-keyword`? There are supposed to be only one toolchain in effect though. I wonder if it would be better to just mutate `toolchain` if `+toolchain` is found without `--toolchain` specified, otherwise it should error out.